### ISSUE TITLE
Why is the default skin 720 only ?

### ISSUE THREAD

**Issue by ian10951** [openatv/enigma2#3412]
This means that if anyone loads a 1080 skin and maybe theres a screen missing, then it reverts to the default skin and everything is the wrong size.

Perhaps its time to update this skin with updates to the text for the colour buttons as thats missing as I've pointed out before and making it both 720 and 1080 compatble.

**Comment by IanSav** [comment]
A better solution is to add the resolution specifications to the default skin.  This will trigger the auto resolution code in the skin.

**Comment by jbleyel** [comment]
Hi @ian10951 , if you find a skinner who can fix / maintain the default skin please let me know.
I have no time for this.

Please note .. the default skin is only for boxes with no extra skin installed or as fallback if your main skin has not all the screens.

Default is probably not the correct name because it's mainly used as emergency / fallback skin.

**Comment by ian10951** [comment]
Yes a thats exactly whats happening, Yesterday I converted my skin to 1080, however I do have a few screens missing as my skin is rather old and Ive been using the default skin as fallback. This was fine when my skin was still 720, but now its 1080 every screen I use the default skin to fallback to is now the incorrect size and very small.

**Comment by jbleyel** [comment]
The default skin is from 1900 and needs a complete refresh.
But this is not that easy and needs a lot of time.

**Comment by ian10951** [comment]
Ok,

Made me laugh that jbleyel, it was funny. lol

**Comment by ian10951** [comment]
> A better solution is to add the resolution specifications to the default skin. This will trigger the auto resolution code in the skin.

Would you be able to do this until the skin can be rewrote ?

Sadly I dont know any skinners , I only know of Kiddac, but I dont know him online personally.

**Comment by kiddac** [comment]
The problem is there is lots of code in the openatv solution that thinks everything is old school sd skins and everything then has to get scaled up with the 1.5 scale factor. . SD boxes are pretty much redundant now. There should be an option for SD and HD default skin.  Metrix HD (which is actually SD unless you change the settings), also causes problems with other plugins due to confusion about screen sizes.. Because all this scaling in the code has always caused lots of problems. 

I have a windows app that will scale up 720 skins. Its very simply to use. 

https://www.linuxsat-support.com/thread/128786-enigma2-1080-skin-converter/

**Comment by IanSav** [comment]
OpenATV does not use simple scale factors.  The scaling done by OpenATV is completely different from all other images.  Scaling can be done dynamically and automatically on a screen by screen basis.  Any screen can be scaled up or down to suit the resolution of the active skin.

The missing part for ian10951 is that the default/emergency skin has not been updated to use this new facility as it has historically been shared with other images.  This is about to change as plans are in place for OpenATV to have its own dedicated default/emergency skin that will be included with the OpenATV image.  This skin will offer automatically scalable default/emergency screens to back up any skin, in any resolution, that is missing those screens.

**Comment by kiddac** [comment]
By sd skin - I mean 1280 x 720,
hd skin - I mean 1920 x 1080

Because that used to be the old way we referred to them whether you had an sd box or a hd box. 

Yes I know technically speaking 
1280 x 720 = hd
1920 x 1080 = fhd
2560 x 1440 = qhd
3840 x 2160 = 4k

**Comment by kiddac** [comment]
And maybe get rid of metrix skin altogether, The design is so 10 years ago, I don't know anyone that uses it in the UK. Maybe they do in Germany???
Just have one nice modern default skin, that doesn't have all those unnecessary large icons.

**Comment by jbleyel** [comment]
I think you are wrong because Metrix is the most used skin of ATV and all clones of ATV and not only in Germany.

**Comment by ian10951** [comment]
Kiddac is correct about Matrix in England.

Absolutely no one uses it that I know of, most find it ugly and very old fashioned. However as stated this might be different in Germany.

**Comment by IanSav** [comment]
What is the most popular skin for OpenATV in England?

Who is the skin author?  Perhaps they would be interested in providing an OpenATV version.

**Comment by ian10951** [comment]
Hard to say. I just know its not Matrix. Skins are a personal choice and many like myself adapt much older skins to run on modern images, simply because there are very few if any new ATV skins. My skin is an old adapted PLi-HD skin and I do know that Pli full night skin thats been taken over by Stein17 on the feeds is popular, mainly due to the fact that its one of the few ATV skins still updated.

**Comment by IanSav** [comment]
Do you want to have a look at OverlayHD and tell me what you think.

Yes it is a biased opinion but I would be interested to know how my skin compares to your expectations.  It is going to get some updates soon to offer all the latest OpenATV skinning features.  If OverlayHD is more to your liking then I would he happy to expand it to cover all the OpenATV screens and plugins that are not currently supported.

**Comment by ian10951** [comment]
Doesnt make a difference to me to be honest as long as its there for other skins to fallback too if they have missing screens, so all possible screens would have to work in both 720 and 1080.

I think this is what 99% of users will use it for. I doubt many will use it as their main skin.

**Comment by ian10951** [comment]
Quite a nice skin that. First time ive ever seen it. Personally I would have blended the infobar rather than having sharpe edges but hey thats just me. It is however just 720 so in its current state it wouldnt be suitable as the fallback/default skin for other skins to fallback to if they have missing screens, because the situation would be exactly as it is now with the existing default skin in the fact that if users were using a 1080 skin, everything would be tiny on any screens their skin had missing and therefore used your skin as fallback.

**Comment by IanSav** [comment]
When the default/emergency skin is rebuilt all the autoscaling code changes will be made so that it can and will offer replacement screens for any skin missing those screens.

As you say, the hard edge in OverlayHD is a design choice.  Did you have a look at the configuration plugin?

**Comment by ian10951** [comment]
Yes the plugin is a nice touch and the skin seems very variable when using the plugin.

It is one of the better skins I've seen. If I used it though I would have to add more info into the infobar, but again thats just me as I like a lot of info about if Exteplayer3 or Gstreamer is in use, VPN status and image information. But again thats just me, everyone is different lol.

**Comment by ian10951** [comment]
This is my InfoBar, its how I like the info shown to me. But I do know its not to everyones taste. 

![5002_0_19_71_B_85_C00000_0_0_0_20240902151223](https://github.com/user-attachments/assets/335a9020-084e-45b1-94f0-b8f0c92e3cab)

**Comment by IanSav** [comment]
You are used to that screen so I am sure you can pick out the information you want.

For me, I think there is too much information and it will be hard to read when you are sitting in a seat a few metres away from the screen.  (That is one of the main reasons why OverlayHD is a HD (720p) skin.)

When I get leave from developing to spend time on OverlayHD I would be happy to take input and suggestion to increase its appeal.  I doubt that I will ever get to the data density that you want but there other ways to skin that cat.  For example adding another InfoBar screen that can provide more of the data you would like to see.

**Comment by IanSav** [comment]
Yes, building skins is very time consuming.  If you try to build them to have pixel perfect alignment then it is even more time consuming.

As I am now primarily a developer so I don't think I will be doing the work on the new default/emergency skin.  I will be a contributor but probably not the main coder.  We will see how the job allocations turn out.  What I can say is that the new skin will be minimum on graphics and built for speed and simplicity.  It will be fully functional and is intended to get the job done but will be minimalistic.

I am hoping that the concept designs are being organised soon and then we will start to flesh it out.  As the skin is built all the embedded screens will be removed from the code into this skin.  This will reduce the size of the code and provide a complete skin which other skinners can use as a sample of what is required together with some ideas on how to get the job done.

Part of my assigned work has been to write some skinner documentation to help skinners understand all the new and enhanced skinning options being made available to OpenATV skinners.  The manual for the new ChannelSelection screen and skins has been written but is being proof read and reviewed before being published.  Other manuals are also planned to be written.

**Comment by IanSav** [comment]
If you are currently having issues maintaining your skin then you should reach out for help in either the forums or in Discord to see if others, including the developers, can help you out.  The developer group is very busy but we may be able to help out if time permits.

The Metrix skin is often updated quite quickly to match the new code.  OverlayHD is also updated fairly promptly as well.  You may care to keep an eye on these skins to give you some hints and clues about how some of the new code works.  I am working on updates for OverlayHD to switch from Python templates to the new XML based skin templates.  I also am preparing the new skin to allow use of the new skinnable ChannelSelection screen.  Unfortunately my development duties are trumping my skinning duties so I can't publish OverlayHD updates as often as I would like.

I also need to find time to publish my InfoBarTimers plugin that adds additional information the the standard InfoBar display.  This plugin might give you some ideas on how to add more information to the standard InfoBar to get someof that extra information you would like to see.

**Comment by IanSav** [comment]
With regard to skin documentation, I suggest you keep an eye on the "doc" directory in the repository.  That is where all the documentation is kept.  This is where the new documentation will be added.

**Comment by kiddac** [comment]
When starting out with skinning in Enigma2, most people, including myself, turn to the default skin to figure out how to create a custom skin. Since there’s no definitive guide or "bible" for Enigma2 skinning, you typically begin by copying and tweaking someone else’s work until it all clicks. However, the problem is that the default skin isn’t given much attention or priority. Instead, all the focus is on the Metrix skin, which is far too complex to learn from. Metrix is a maze of templates within templates, and its customizability only adds to the confusion.

As Ian pointed out, when something breaks, skinners often refer to the image's default skin to troubleshoot and adjust their code. But if the default skin itself isn’t fully functional, it makes things even more challenging.

A default skin should be clean, simple, and include all the necessary elements. It should avoid large, unsightly icons and multiple layouts. More comments explaining what each widget line does and where to find its source on GitHub would also be helpful.

Skinning in 2024 is easier because design trends have moved towards simplicity and flat aesthetics. Glossy effects are unnecessary, shadows can be skipped, and colour palettes can be limited. Subtle gradients work well, but strong gradients and unnecessary graphics should be avoided.

**Comment by jbleyel** [comment]
We have already started a skin documentation.
You can help with review if you like.

**Comment by kiddac** [comment]
I am trying to retire. My days of coding for enigma2 are coming to an end. Very soon. It is someone else's turn to take up the challenge. Which is why I am trying to get everything fixed before I disappear. :)

**Comment by jbleyel** [comment]
> I am trying to retire. My days of coding for enigma2 are coming to an end. Very soon. It is someone else's turn to take up the challenge. Which is why I am trying to get everything fixed before I disappear. :)

Unfortunately another one less, but you may enjoy your retirement.
Let's see if someone can keep your code up to date and fix any bugs.

**Comment by atvcaptain** [comment]
This has changed with OpenATV 8.0. Users can now switch the skin resolution between HD and FHD without restarting the GUI, and WQHD support will follow soon.

We deliberately continue to use HD as the default because we still support older receivers that cannot properly handle FHD skins. Users with suitable hardware can switch to FHD with a single click.

Both Metrix and Umbra support this resolution switching. Third-party skins can also be provided in the appropriate resolutions.

For these reasons, we currently see no need to force FHD as the default resolution for all receivers.

**Comment by IanSav** [comment]
HD skins are also smaller on disk and faster to load and process.  I still also believe that when sitting on a couch, well away from the TV screen, there is little extra value in going to FHD or higher resolution screens.  Are you there to watch the UI or the actual event on the TV?   ;)

**Comment by ian10951** [comment]
@IanSav 

Your completely missing the point.

No one uses the python default skin as their main skin, so you comments above although true are not valid here.

The clue is in the name. The fallback skin is there so if any FHD or HD skins have missing screens then it fallsback to that skin. The trouble is with it being just 720 that anyone using a 1080 skin who has a missing screen will see tiny characters and fonts. The python fallback skin should be both 720 and 1080.

**Comment by IanSav** [comment]
@ian10951,

With all due respect, I don't think I am missing the point at all.  My recent comments reflect that while the desire to create and use higher resolution skins is prevalent I still believe they have questionable real-world benefits.

With regard to the original comments regarding the extremely outdated "skin-default" skin, you have no disagreement from me.  I have elected to create a new DefaultHD skin that will become the replacement of "skin_default" and will be directly linked to OpenATV and will hopefully be kept in sync with all the OpenATV development and enhancements.

The current issue is that OpenATV development, enhancement and testing for OpenATV 8.0 is taking considerable developer effort.  As a code and skin developer my resources are currently overstretched just attempting to cope with the coding and testing side and the skinning side is not getting the attention it needs.

There is another related issue that I also should highlight.  As a result of all the new code changes there is a significant impact on the skins needed for OpenATV.  It is unwise to devote too much effort on skinning screens when the code that needs to be skinned is still evolving and changing.

DefaultHD will come but it will take some time.
